### DISCUSSION TITLE
Do I need to focus Blazor development on WASM?

### DISCUSSION THREAD

**Discussion by georgeemr** [dotnet/aspnetcore#29408] [General]
Hi, We are going to develop 3 big projects for my work and we want the projects to be SPA. For this, we are evaluating the frameworks that are available right now.
We are all Dot Net developers and we always make our projects in NET CORE MVC.
With Blazor we can determine that we can skip frameworks like Angular, React, or Vue but we don't want it to be WASM, for security reasons we need the code to run on the server, plus we don't like the download size that takes a WASM app to download to the client versus.

Do you recommend Blazor Server for big projects? These projects actually are for public users and not for internal users of our organization, I mean, one of our questions are if in the future, Blazor will focus only on WASM and not on Server production? As we are going to develop this with Blazor Server it's going to be sad if in a few years we will be obligated to convert the project to wasm.

**Comment by Xyncgas** [comment]
You can call api from app to server still
just use it like a javascript engine

Sounds like you are starting to use this thing, I say blazor has components that already comes with it and you can make your own, but you need to find blazor libraries to make more UI if you want to do it fast and don't want to write the code yourself meanwhile javascript UI framework is everywhere

**Comment by jeanpoulin** [comment]
I'd say use WebAssembly but Hosted. So far, Wasm Standalone like I used can give you head ache to make some Login security especially with social network external login like google and Facebook.  With .Net 8 most of drawback of using WASM will be mitigated while keeping advantage of Wasm.
